### PR TITLE
a warning is displayed if you upload a png encouraging grid use

### PR DESCRIPTION
**Pull request by NickPapacostas**
@akash1810 
@Reettaphant 
@rrees 

PNG's going out today, will be nice to have this sitting in the wings ready to soar 🐦 

Right now it still let's them go through with the upload, I figure there may be some extra-ordinary circumstances where we don't want to _prevent_ the uploader from uploading right?

s3-uploader-png-warning

### CODE REVIEW

File: app/views/index.scala.html [hunk 1]
```diff
@@ -6,6 +6,9 @@
     <div class="mdl-grid">
         <div class="mdl-cell mdl-cell--12-col">
             <h1>Choose files to upload</h1>
+            <div class="upload-error hidden" id="png-error">
+                <p>The Grid now accepts PNGs! <a href="https://media.gutools.co.uk/upload">Give it a shot</a></p>
```
Comment (akash1810) [inline app/views/index.scala.html]: we tend not to put internal URLs in public repos...

File: app/views/index.scala.html [hunk 2]
```diff
@@ -6,6 +6,9 @@
     <div class="mdl-grid">
         <div class="mdl-cell mdl-cell--12-col">
             <h1>Choose files to upload</h1>
+            <div class="upload-error hidden" id="png-error">
```
Comment (akash1810) [inline app/views/index.scala.html]: `hidden` would be `upload-error--hidden` for BEM

File: public/js/upload.js [hunk 3]
```diff
@@ -9,8 +9,15 @@ function handleFileSelect(event) {
...
     const selected = document.querySelector("#selectedFiles ul");
     selected.innerHTML = "";
+
     for(var i=0; i<files.length; i++) {
-        const file = files[i];
-        selected.innerHTML += "<li>" + file.name + "<br/>" + "</li>";
+      var file = files[i];
+      var ext = file.name.match(/\.(.+)$/)[1];
+      if(ext == "png"){
+        newClassName = document.querySelector("#png-error").className.replace("hidden", "");
```
Comment (akash1810) [inline public/js/upload.js]: `classList.remove` rather than `replace`?

File: public/js/upload.js [hunk 4]
```diff
@@ -9,8 +9,15 @@ function handleFileSelect(event) {
     const files = event.target.files;
     const selected = document.querySelector("#selectedFiles ul");
     selected.innerHTML = "";
+
     for(var i=0; i<files.length; i++) {
-        const file = files[i];
-        selected.innerHTML += "<li>" + file.name + "<br/>" + "</li>";
+      var file = files[i];
```
Comment (akash1810) [inline public/js/upload.js]: const?

File: public/js/upload.js [hunk 5]
```diff
@@ -9,8 +9,15 @@ function handleFileSelect(event) {
...
+      var ext = file.name.match(/\.(.+)$/)[1];
+      if(ext == "png"){
+        newClassName = document.querySelector("#png-error").className.replace("hidden", "");
+        document.querySelector("#png-error").className = newClassName;
+      }
+      selected.innerHTML += "<li>" + file.name + "<br/>" + "</li>";
```
Comment (akash1810) [inline public/js/upload.js]: minor: [`appendChild`](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) is preferable over `innerHTML`

**Comment by akash1810** [comment]
Not convinced this is needed - jpgs are supported in Grid and we don't warn for those files... Maybe sending comms about Grid's new PNG support is enough?

File: app/views/index.scala.html (hunk 2 above)
Comment (NickPapacostas) [inline app/views/index.scala.html]: I'm using the css libraries 'hidden' class. 

i guess i'll make it upload-error upload-error--hidden ?

**Comment by NickPapacostas** [comment]
I see what you mean @akash1810 I guess the difference is that previously the _only_ way to upload pngs was through this tool, and now that's not the case. Whereas with jpegs, the grid has always been the go-to solution. 

@rrees initially suggested it to me.

**Comment by rrees** [comment]
@akash1810 at some point we need to resolve why people are uploading JPGs via the uploader as well.

**Comment by rrees** [comment]
👍

**Comment by akash1810** [comment]
A thought that is indirectly related to this PR...

Would it be a good idea to (for files that Grid supports) push files uploaded to S3 Uploader to the S3 Watcher's bucket too? That way they are loaded into the Grid at the same time.

**Comment by kenoir** [comment]
Nice.

On Wed, 25 May 2016 at 10:27, Akash A <notifications@github.com> wrote:

> A thought that is indirectly related to this PR...
>
> Would it be a good idea to (for files that Grid supports) push files
> uploaded to S3 Uploader to the S3 Watcher's bucket too? That way they are
> loaded into the Grid at the same time.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/s3-upload/pull/12#issuecomment-221611617>
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Comment by NickPapacostas** [comment]
@akash1810 hmmm maybe we could also make the links on the post-upload page be grid links ^_^

**Comment by paperboyo** [comment]
@akash1810 Great idea! From discussion on Slack:
> Grid may be down, but the images should wait somewhere to be recovered/ingested by Grid. Uploader does not let one enter metadata - but upon resurrection maybe Grid could pull metadata from what user entered in Composer for these s3-uploader images. We should even know the uploader (person) to backfill Grid's meta...

**Pull request merged**